llvm/utils/TableGen/GlobalISelEmitter.cpp

```diff
@@ -1412,15 +1412,18 @@ Expected<BuildMIAction &> GlobalISelEmitter::createAndImportInstructionRenderer(
   action_iterator InsertPt = InsertPtOrError.get();
   BuildMIAction &DstMIBuilder = *static_cast<BuildMIAction *>(InsertPt->get());
 
-  for (auto PhysInput : InsnMatcher.getPhysRegInputs()) {
-    InsertPt = M.insertAction<BuildMIAction>(
-        InsertPt, M.allocateOutputInsnID(),
-        &Target.getInstruction(RK.getDef("COPY")));
-    BuildMIAction &CopyToPhysRegMIBuilder =
-        *static_cast<BuildMIAction *>(InsertPt->get());
-    CopyToPhysRegMIBuilder.addRenderer<AddRegisterRenderer>(
-        Target, PhysInput.first, true);
-    CopyToPhysRegMIBuilder.addRenderer<CopyPhysRegRenderer>(PhysInput.first);
+  for (auto PhysOp : M.physoperands()) {
+    auto &OpInsnMatcher = PhysOp.second->getInstructionMatcher();
+    for (auto PhysInput : OpInsnMatcher.getPhysRegInputs()) {
+      InsertPt = M.insertAction<BuildMIAction>(
+          InsertPt, M.allocateOutputInsnID(),
+          &Target.getInstruction(RK.getDef("COPY")));
+      BuildMIAction &CopyToPhysRegMIBuilder =
+          *static_cast<BuildMIAction *>(InsertPt->get());
+      CopyToPhysRegMIBuilder.addRenderer<AddRegisterRenderer>(
+          Target, PhysInput.first, true);
+      CopyToPhysRegMIBuilder.addRenderer<CopyPhysRegRenderer>(PhysInput.first);
```

Comment:
So looks `CopyPhysRegRenderer`, it could be just `CopyRenderer`.

Reply:
It's not possible, `CopyRenderer` consumes `StringRef SymbolicName` when we have `Record *`. It looks like `CopyPhysRegRenderer(unsigned NewInsnID, const Record *Reg)` is the right choice here.

Reply:
`SymbolicName` is used for debugging purposes. It should be `Reg->getName()` (see `CopyPhysRegRenderer::emitRenderOpcodes`).